### ISSUE TITLE
Order of command args too strict

### ISSUE THREAD

**Issue by sfozz** [hadolint/hadolint#896]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

Order of files to parse and arguments should be changed so that you can specify optional arguments to hadolint when running in pre-commit.

### Actual behavior

the following error is shown when scanning a container that and we want to set the failure threshold to error 

```
Lint Dockerfiles.........................................................Failed
- hook id: hadolint-docker
- duration: 0.71s
- exit code: 1

Invalid option `--failure-threshold error'

Usage: hadolint [-v|--version] [-c|--config FILENAME] [DOCKERFILE...]
                [--file-path-in-report FILEPATHINREPORT] [--no-fail]
                [--no-color] [-V|--verbose] [-f|--format ARG] [--error RULECODE]
                [--warning RULECODE] [--info RULECODE] [--style RULECODE]
                [--ignore RULECODE]
                [--trusted-registry REGISTRY (e.g. docker.io)]
                [--require-label LABELSCHEMA (e.g. maintainer:text)]
                [--strict-labels] [--disable-ignore-pragma]
                [-t|--failure-threshold THRESHOLD]
  Lint Dockerfile for errors and best practices
```

Under pre-commit we have no control over how the args are passed to hadolint, so what happens is that the `--failure-threshold error` is specified before the list of docker files. As shown by the following strace output

```
[pid 400697] execve("/usr/bin/docker", ["/usr/bin/docker", "run", "--rm", "-u", "1000:1000", "-v", "/home/sfozz/src/site"..., "--workdir", "/src", "ghcr.io/hadolint/hadolint:v2.9.3", "hadolint", "--failure-threshold error", "api/Dockerfile", "client/Dockerfile", "pdf/Dockerfile"], 0xfa18d0 /* 67 vars */ <unfinished ...>
```

If I execute the same command manually but put the optional argument after the dockerfiles then this works as expected.

### Steps to reproduce the behavior

Create a Dockerfile with warning class error

`.pre-commit-config.yaml` at base of repo
```yaml
repos:
  - repo: https://github.com/hadolint/hadolint
    rev: v2.10.0
    hooks:
      - id: hadolint-docker
        verbose: true
        args:
          - --failure-threshold error
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**
Haskell Dockerfile Linter 2.9.3

```bash
(paste your output here)
```

**Dockerfile** (if relevant)

```Dockerfile
(paste content of your Dockerfile)
```

**Additional environment details (OS, stack version, etc.)**

Ubuntu 20.04 LTS, pre-commit 2.13.0

**Comment by m-ildefons** [comment]
Hey Steve,

in your `.pre-commit-config.yaml` file, the extra arguments to Hadolint should be separate elements in the yaml array `args`. You can see in the `strace` output that the `execve` call groups the two arguments `--failure-threshold` and `error` into a single string which contains a space. That is wrong, it should be two different strings. You can achieve that by specifying the `args` in the `.pre-commit-config.yaml` like so:

```yaml
repos:
  - repo: https://github.com/hadolint/hadolint
    rev: v2.10.0
    hooks:
      - id: hadolint-docker
        verbose: true
        args:
          - --failure-threshold
          - error
```

I have no idea why it works when the order of the arguments is changed though

**Comment by sfozz** [comment]
:facepalm: should have thought of that... Working now.

Many thanks
